include/h2o.h

```diff
@@ -1880,6 +1890,16 @@ inline void *h2o_context_get_logger_context(h2o_context_t *ctx, h2o_logger_t *lo
     return ctx->_module_configs[logger->_config_slot];
 }
 
+inline void h2o_context_get_storage_index(h2o_context_t *ctx, size_t *key)
+{
+    if (*key == SIZE_MAX)
+        *key = ctx->storage.size;
+    if (ctx->storage.size <= *key) {
+        h2o_vector_reserve(NULL, &ctx->storage, *key + 1);
```

Comment:
Maybe you need to zero-clear the allocated memory; i.e. `memset(ctx->storage.entries + ctx->storage.size, 0, (*key + 1 - ctx->strorage.size) * sizeof(ctx->storage.entries[0]);`.